File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    for a in arr:
        try:
            if answer[-1] == a:
                continue
            else:
                answer.append(a)
        except:
            answer.append(a)
    return answer
# def solution(s):
#     a = []
#     for i in s:
#         if a[-1:] == [i]: continue
#         a.append(i)
#     return a

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public Object[] solution(int []arr) {
        // Object[] answer = {};
        Stack<Integer> s = new Stack<>();
        for (int a: arr) {
            if (s.isEmpty()) {
                s.add(a);
            }
            else {
                if (s.peek() == a) {
                    continue;
                } else {
                    s.add(a);
                }
            }
        }
        // answer = s.toArray();
        return s.toArray();
    }
}